src/main/java/uk/gov/hmcts/reform/cpo/repository/CasePaymentOrdersRepositoryImpl.java

```diff
@@ -45,11 +44,30 @@ public void deleteByCaseIds(List<Long> caseIds) {
     }
 
     private void validateAllEntriesExistByCaseIds(List<Long> caseIds) {
+        List<String> nonExistentCaseIds = new ArrayList<>();
         for (Long cid : caseIds) {
             if (casePaymentOrdersJpaRepository.findAllByCaseId(cid).isEmpty()) {
-                throw new CasePaymentOrderCouldNotBeFoundException(ValidationError.CPO_NOT_FOUND_BY_CASE_ID);
+                nonExistentCaseIds.add(String.valueOf(cid));
+            }
+        }
+        throwExceptionIfCposNotFound(nonExistentCaseIds);
+    }
+
+    private void validateAllEntriesExistByUuid(List<UUID> uuids) {
+        List<String> nonExistentUuids = new ArrayList<>();
+        for (UUID uuid : uuids) {
+            if (casePaymentOrdersJpaRepository.findAllById(uuid).isEmpty()) {
```

Comment:
Same as above query comment. Also check if there is a better way to avoid multiple DB calls.

Reply:
Fixed in https://github.com/hmcts/cpo-case-payment-orders-api/pull/82/commits/7503751b934686d8ada84a8f2ab06bcb589e7bf5